Move modules into proper folder structure and rename to proper names

Berthalamew opened the pull request:
- try and organize some of the Engine related code we have into the Blam/Engine folder instead 
- rename some structs so that they parody the original struct names
- move some of the structs/definitions to proper modules

Berthalamew committed:
rename Actor header to actors

update logic that uses the struct

Berthalamew committed:
Decapitalize Memory folder

Berthalamew committed:
Rename TagDataIterator to cache_files

Berthalamew committed:
Rename DataArray header to data

Berthalamew committed:
Remove capitalization from game folder

update header references

Berthalamew committed:
Remove capitalization from Objects and modules inside of it

update header referemces in other modules

Berthalamew committed:
Update filenames in vcxproj

Berthalamew committed:
Move all non game related modules out of game folder

update header references

Berthalamew committed:
Move skull code to cheats module

Berthalamew committed:
Uncapitalize modules in game folder

update references to headers in other modules

Berthalamew committed:
Move s_aim_assist_targetting_data into aim_assist header

Berthalamew committed:
Move players and player_control into proper folder

update references to headers

Berthalamew committed:
Move LocalPlayers logic into players

update references to header

Berthalamew committed:
Move s_player_actions out of PlayerActions module

- rename s_player_actions to player_actions since it's the original variable name
- update references to header

Kantanomo commented:
Waiting for appvayor.

Kantanomo commented on xlive/Blam/Engine/Simulation/SimulationWorld.h:
player_action is not just used for Simulation world, it's used for player control globals. should remain outside of it.

Berthalamew replied:
changed

Kantanomo commented on xlive/Blam/Engine/cache/cache_files.h:
Explain the need for this change.

Berthalamew replied:
it's the module where the tag iterator should be located in. (if there's a better spot let me know)

Kantanomo replied:
why name it cache_files when tag data iterator is also usable inside the memory.

Berthalamew replied:
Since that's where the tag files are being loaded from initially

Kantanomo commented on xlive/Blam/Engine/game/players.h:
Might be fine but it might be better to put them into a Local players structure, local players contains more information. unsure though

Kantanomo replied:
they do resolve to the game state player object but would we rather want for example... local_player(1)->get_gamestaete or game_state->get_local_player(1)

Berthalamew replied:
Local players should be similar enough to players although you're right about extra data at the end. Depending on how different and how much extra data there is it could be fine to just make it a separate module. 
For now it should be fine to keep it within players?

Berthalamew replied:
created the module for local_players

Berthalamew committed:
Move player_action into player_control module

Kantanomo commented on xlive/Blam/Engine/memory/data.h:
i'd prefer data_array unsure though

Berthalamew replied:
data_array is just one part of it and there's other functions in that module that use the data_array for allocating memory

Kantanomo reviewed (changes requested):
Address the comments please, there are some changes that need to be modified and other things need to be explained or reasoned for.

Kantanomo commented:
Changes are purely organizational (renaming, relocating) for easier navigation should not cause any issues in project however the specifics of things moved around should have minimal testing to make sure things still function as they should @Berthalamew .

nukeulater commented:
also https://github.com/pnill/cartographer/pull/413#issuecomment-1467589003

Berthalamew committed:
create local_players module

- rename ENGINE_MAX_PLAYERS to k_maximum_players
- rename ENGINE_MAX_LOCAL_PLAYERS to k_number_of_controllers

Kantanomo commented:
Write a detailed description of the changes and how they will be beneficial to the project and merge it in with that as the comment.,

Berthalamew commented:
Organize some of the Engine related code we have into the Blam/Engine folder and rename the modules to their original counterparts.
This is beneficial since it keeps everything organized and tries to parody the original source layout of halo 2.
